src/libcore/cmp.rs

```diff
@@ -190,8 +245,48 @@ impl Ordering {
 /// - total and antisymmetric: exactly one of `a < b`, `a == b` or `a > b` is true; and
 /// - transitive, `a < b` and `b < c` implies `a < c`. The same must hold for both `==` and `>`.
 ///
+/// ## Derivable
+///
 /// This trait can be used with `#[derive]`. When `derive`d, it will produce a lexicographic
 /// ordering based on the top-to-bottom declaration order of the struct's members.
+///
+/// ## How can I implement `Ord`?
+///
+/// `Ord` requires that the type also be `PartialOrd` and `Eq` (which requires `PartialEq`).
+///
+/// Then you must define an implementation for `cmp`. You may find it useful to use
+/// `cmp` on your type's fields.
+///
+/// Here's an example where you want to sort people by height only, disregarding `id`
+/// and `name`:
+///
+/// ```
+/// use std::cmp::Ordering;
+/// #[derive(Eq)]
+/// struct Person {
+///     id: u32,
+///     name: String,
+///     height: u32,
+/// }
+///
+/// impl Ord for Person {
+///     fn cmp(&self, other: &Self) -> Ordering {
```

Comment:
Yeah I think I kept it `Self` because the trait signature uses it, but it has to and this doesn't have to. 

I feel like preferring self over the class name this is a style thing I got from Greg Brown, with the reasoning that if you change the class name, you'd have fewer places to change. But I don't feel super strongly about it because 1. how often do you change class names anyway and 2. shouldn't your tools (find and replace at least, the compiler, IDE refactoring tools, etc) be handling this for you... so I'm adding a commit changing `&Self` to `&Person` here :)